### DISCUSSION TITLE
Best practice for getting the current locale in a server component

### DISCUSSION THREAD

**Discussion by philip-akerfeldt** [vercel/next.js#58730] [Help]
### Summary

I have a question for the community concerning the best way to access the current page locale from a server component.

In my current project I am using next 14.0.0 with app router and using the [documented example for handling internationalization](https://nextjs.org/docs/app/building-your-application/routing/internationalization).

I have an example page that roughly looks like this:

`page.tsx`
```
type Props = {
  params: { lang: string };
  searchParams: { [key: string]: string | string[] | undefined };
};

async function ConsultantListPage({ params, searchParams }: Props) {

    return (
         <div>
             <h1>Page Title</h1>
             <div >
               <ExampleComponent />
             </div>
         </div>
    )
}
```

With the component looking roughly like this:

`component.tsx`
```
const ExampleComponent = async () => {

    const dict = getDictionary(lang);

    return (
        <div>
            <h1>Component Title</h1>
        </div>
    )
}
```

What I would like to achieve is to be able to get the dictionary in my example component given the current locale.

Which is the preferred way of doing this? 
Passing lang as a prop to the example component?
Getting the dictionary in the page and passing the dictionary as a prop down to the component?

Or is there some other neat way of doing this without having to pass props down to the server component?

I have tried to find a nice solution to this but I have yet to find anything to my liking. Any ideas on how to solve this in a neat way?

Thanks

**Comment by GaziRiad** [comment]
Yeah, i usually do internationalization using third party libraries, not following next's docs and it's much easier. Still a client gave me a project using such config for internationalization and i looking for solution to not pass nested props deep in the component tree 🥲
